libs/cardano-ledger-binary/test/Test/Cardano/Ledger/Binary/Failure.hs

```diff
@@ -36,12 +58,16 @@ prop_shouldFailMapWithDupKeys =
         (property . isLeft) (decode (natVersion @9) encodedMap :: Either DecoderError (Map Int Int))
     )
 
-decode :: DecCBOR a => Version -> Encoding -> Either DecoderError a
-decode version enc =
-  let encoded = serialize version enc
-   in decodeFull version encoded
+-- | Starting in version 9, do not accept duplicates in CBOR sets
+prop_shouldFailSetWithDupKeys :: Property
+prop_shouldFailSetWithDupKeys =
+  forAllBlind genDuplicateListEncoding $
+    ( \setEncoder ->
+        (property . isLeft) (decode (natVersion @9) setEncoder :: Either DecoderError (Set Int))
+    )
```

Comment:
`embedTripRangeFailureExpectation` function from #3332 will produce better error reporting and will ensure this property holds true for future protocol versions
Also parenthesis are redundant.

```suggestion
     \setEncoding ->
      let trip = Trip id (decCBOR @(Set Int))
      in property $ embedTripRangeFailureExpectation trip (natVersion @9) maxBound setEncoding
```